### ISSUE TITLE
Why do ‘<sos>’  and '<eos>' have the same label?

### ISSUE THREAD

**Issue by ZhengkunTian** [vrenkens/nabu#24]
Hi Vincent,
    First of all thank you for your work on this framework, I am learning Seq2Seq for ASR.  I am so confused about **why you use the same label to represent  '<sos>' and '<eos>'?** 
in nabu/neuralnetworks/models/ed_decoders/rnn_decoder.py
`expanded_targets = tf.pad(targets.values()[0], [[0, 0], [1, 0]],
                                  constant_values=output_dim-1)`
in nabu/neuralnetworks/trainers/loss_functions.py  (cross_entropy_eos)
`final_targets = tf.tile([output_dim-1], [batch_size])`

what differences using two labels and using the same label have? 
     Thank you and have a good day!

Damon

**Comment by vrenkens** [comment]
There is no big differnce. I guess you can do both. I chose to use the same label because otherwise the sos label would never be decoded by the decoder and thus the corresponding weights in the softmax would never get gradients and that is a bit of a waste (same thing for the eos as input).

**Comment by ZhengkunTian** [comment]
Thanks for your answer! It's very helpful!
